src/pathfinding_service/service.py

```diff
@@ -98,6 +98,8 @@ def __init__(  # pylint: disable=too-many-arguments
             message_received_callback=self.handle_message,
             address_reachability_changed_callback=self.handle_reachability_change,
         )
+        self.matrix_listener.startup_finished.wait(timeout=1)
+        log.debug("Started Matrix client")
```

Comment:
Ideally, creating a `PathfindingService` would not run anything until `_run` is called. Thus I would expect the `wait` below `self.matrix_listener.start()`.

If we don't want to clean that up right now, this change is the right one.

Reply:
You're right, it's still morning.